library/std/src/sys/unix/thread.rs

```diff
@@ -285,17 +285,20 @@ impl Drop for Thread {
 }
 
 #[cfg(any(target_os = "linux", target_os = "macos", target_os = "ios", target_os = "watchos"))]
-fn truncate_cstr(cstr: &CStr, max_with_nul: usize) -> crate::borrow::Cow<'_, CStr> {
-    use crate::{borrow::Cow, ffi::CString};
-
-    if cstr.to_bytes_with_nul().len() > max_with_nul {
-        let bytes = cstr.to_bytes()[..max_with_nul - 1].to_vec();
-        // SAFETY: the non-nul bytes came straight from a CStr.
-        // (CString will add the terminating nul.)
-        Cow::Owned(unsafe { CString::from_vec_unchecked(bytes) })
-    } else {
-        Cow::Borrowed(cstr)
-    }
+fn truncate_cstr<const MAX_WITH_NUL: usize>(cstr: &CStr) -> [libc::c_char; MAX_WITH_NUL] {
+    let bytes = cstr.to_bytes_with_nul();
+    let len = usize::min(bytes.len(), MAX_WITH_NUL - 1);
+
+    // SAFETY: All valid values of &[u8] are also valid values of &[i8].
+    // i8 == c_char for all platforms for which this function is compiled, and fail
+    // to compile otherwise.
+    let bytes = unsafe { crate::mem::transmute::<&[u8], &[i8]>(bytes) };
+
+    // We are writing at most (MAX_WITH_NUL - 1) bytes to result, so the resultant buffer is
+    // guaranteed to be NUL terminated.
+    let mut result = [0_i8; MAX_WITH_NUL];
```

Comment:
This needs to be an array of `c_char` too, since different targets may be aliased to `i8` or `u8`.

```suggestion
    let mut result: [libc::c_char; MAX_WITH_NUL] = [0; MAX_WITH_NUL];
```

Reply:
Resolved via type inference.